File: abelkhan_rpc_demo/proto/hub_call_client/csharp/caller/hcallccaller.cs
/*this caller file is codegen by abelkhan for c#*/
using System;
using System.Collections;
using System.IO;

namespace ntf
{
    public class hcallc
    {
        public hcallc()
        {
        }

        public hcallc_cliproxy get_client(string uuid)
        {
            return new hcallc_cliproxy(uuid);
        }

        public hcallc_cliproxy_multi get_multicast(ArrayList uuids)
        {
            return new hcallc_cliproxy_multi(uuids);
        }

        public hcallc_broadcast get_broadcast()
        {
            return new hcallc_broadcast();
        }

    }

    public class hcallc_cliproxy
    {
        private string uuid;

        public hcallc_cliproxy(string _uuid)
        {
            uuid = _uuid;
        }

        public void hcallc(String argv0)
        {
            hub.hub.gates.call_client(uuid, "hcallc", "hcallc", argv0);
        }

    }

    public class hcallc_cliproxy_multi
    {
        private ArrayList uuids;

        public hcallc_cliproxy_multi(ArrayList _uuids)
        {
            uuids = _uuids;
        }

    }

    public class hcallc_broadcast
    {
        public hcallc_broadcast()
        {
        }

    }

}

File: abelkhan_rpc_demo/proto/hub_call_client/js/caller/hcallccaller.js
/*this ntf file is codegen by ablekhan for js*/

function hcallc(hub_ptr){
    this.get_client = function(uuid){
        return new hcallc_cliproxy(uuid, hub_ptr);
    }
    this.get_multicast = function(uuids){
        return new hcallc_cliproxy_multi(uuids, hub_ptr);
    }
    this.get_broadcast = function(){
        return new hcallc_broadcast(hub_ptr);
    }
}
module.exports.hcallc = hcallc;
module.exports.hcallc_cliproxy = hcallc_cliproxy;
module.exports.hcallc_cliproxy_multi = hcallc_cliproxy_multi;
module.exports.hcallc_broadcast = hcallc_broadcast;

function hcallc_cliproxy(uuid, hub_ptr){
    this.uuid = uuid;

    this.hcallc = function(argv0){
        hub_ptr.gates.call_client(uuid, "hcallc", "hcallc", argv0);
    }
}
function hcallc_cliproxy_multi(uuids, hub_ptr){
    this.uuids = uuids;
}
function hcallc_broadcast(hub_ptr){
}
